hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -2821,7 +2822,23 @@ else if ( isAllOrDirtyOptLocking() && oldFields != null ) {
 					String[] propertyColumnWriters = getPropertyColumnWriters( i );
 					boolean[] propertyNullness = types[i].toColumnNullness( oldFields[i], getFactory() );
 					for ( int k = 0; k < propertyNullness.length; k++ ) {
-						if ( propertyNullness[k] ) {
+						if ( propertyColumnNames[k] == null && propertyColumnFormulaTemplates[k] != null ) {
```

Comment:
Same issue as below, the first index to use for `propertyColumnFormulaTemplates` should be the property index.

```suggestion
						if ( propertyColumnNames[k] == null && propertyColumnFormulaTemplates[i] != null ) {
```

Reply:
You are right, thanks @beikov I'm going to fix it